docs/config/index.md

```diff
@@ -427,6 +427,18 @@ export default async ({ command, mode }) => {
 
   File system watcher options to pass on to [chokidar](https://github.com/paulmillr/chokidar#api).
 
+### server.fsServeRoot
+
+- **Type:** `string`
+
+  Restrict files that could be served via `/@fs/`. Accessing files outside this directory will result in a 403.
+  
+  Vite will search for the root of the potential workspace and use it as default. A valid workspace met the following conditions, otherwise will fallback to the [project root](/guide/#index-html-and-project-root).
+  - contains `workspaces` field in `package.json`
+  - contains one of the following directory/file
+    - `.git`
```

Comment:
I don't know if we should add `.git` to define a workspace. I understand that this will cover most use cases, as almost everyone is using git. But I think it could be quite surprising that a project works when it is in a git repo, but not when it is copied locally using degit for example. Could you explain a bit more the rationale?

Reply:
The reason I add it as I was thinking to be as safer as possible, having '.git' allows the detection to be returned earlier if matched. Yeah, agree with your points as well, we can remove it for now I guess